exporter/datasetexporter/datasetexporter.go

```diff
@@ -117,3 +118,31 @@ func updateWithPrefixedValues(target map[string]interface{}, prefix string, sepa
 
 	}
 }
+
+func inferServerHost(
+	resource pcommon.Resource,
+	attrs map[string]interface{},
+	hostSettings ServerHostSettings,
+) string {
+	// first use value from the attribute serverHost
+	valA, okA := attrs[add_events.AttrServerHost]
+	if okA {
+		host := fmt.Sprintf("%v", valA)
+		if len(host) > 0 {
+			return host
+		}
+	}
+
+	// then use value from resource attributes - serverHost and host.name
+	for _, resKey := range []string{add_events.AttrServerHost, "host.name"} {
+		valR, okR := resource.Attributes().Get(resKey)
+		if okR {
+			host := valR.AsString()
+			if len(host) > 0 {
+				return host
+			}
+		}
+	}
+
+	return hostSettings.ServerHost
```

Comment:
It looks like if we have the default config and no attribute or resource attribute is set for the server host we just return whatever the config specified as the `server_host`. Shouldn't we only return `hostSettings.ServerHost` if it's non-empty, and then if it's empty return the node's host name if `use_host_name` is true?

I might have missed some functionality somewhere, so let me know if that's the case. We should add a test as well to make sure we have the default case covered and the `serverHost` gets set properly.

Reply:
> Shouldn't we only return hostSettings.ServerHost if it's non-empty, and then if it's empty return the node's host name if use_host_name is true?

Yeah, that seems like a reasonable approach to me, but will let @martin-majlis-s1 confirm.

Reply:
I have just changed it! My initial thinking, that since the logic is implemented in the library as well, there is no point to implement it here as well. But you were right, that it's better when the exporter fills the structures properly.